### ISSUE TITLE
Gameweek Data gets out of alignment

### ISSUE THREAD

**Issue by mpgallagher** [H-Cox/FPL#5]
Hey Henry,

I noticed an issue where the alldata.csv output file has it's gameweek history out of synch after a few rows. An example would be for you to look at Mane's (ID: 251) Gameweek 1 ICT Index. It's significantly lower than his basic data value (18.5)

I'm sorry that I'm just learning python as I go, so you may find the solution quicker than I would. I'll append this issue if I find the problem. Feels like somewhere the playerid's are getting remapped when adding in the gameweek history.

**Comment by sj8070** [comment]
Yep i noticed this and built an algo to re map them. shout if anyone needs

**Comment by jeski** [comment]
I'd be interested if there isn't a pull/merge

**Comment by mpgallagher** [comment]
looks like there is an iteration through basic_data, with an assumption that the row would match the playerid. However, basic_data isn't pulled in playerid order.

+1 on jeski's comment.

**Comment by mpgallagher** [comment]
I may have fixed this locally by changing:
(Starting line 345)
                # data is imported here
                temp_data = import_player_data2(i+1,team_name_data)

to

		# data is imported here
		row_player_id = basic_data_values[i][0]
		temp_data = import_player_data2(row_player_id,team_name_data)



Of course, if I knew more about pulling/editing/merging in github, I'd submit it myself. Maybe by day's end.

**Comment by jeski** [comment]
fork the repo. create a branch, check your changes into the branch. push your changes. make a pull request.

https://help.github.com/articles/creating-a-pull-request/

**Comment by H-Cox** [comment]
Hi everyone,

Thanks very much for the comments and suggestions. I am really busy at the moment with work and stuff so don't get much time for messing around with FPL data sadly.

I have implemented the change suggested by mpgallagher so hopefully now you can pull the repository and all the rows should be in the correct order. Although I haven't tested this.

Over the next few weeks I hope to be totally rewriting this project using a lot less lines of code and hopefully also implementing a testing framework so that we can quickly run tests to ensure everything works as expected.

If you guys wanna get onboard and help with that then it would be great and a nice little open source project for us all.

Cheers,

Henry
